akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -142,7 +143,7 @@ private[remote] final class HeaderBuilderImpl(val compression: LiteralCompressio
 
   def senderActorRef_=(ref: ActorRef): Unit = {
     _senderActorRefIdx = compression.compressActorRef(ref)
-    if (_senderActorRefIdx == -1) _senderActorRef = ref.path.toSerializationFormat
+    if (_senderActorRefIdx == -1) _senderActorRef = Serialization.serializedActorPath(ref) // includes local address from `currentTransportInformation`
```

Comment:
`toSerializationFormat` does not include the address from the `currentTransportInformation`, one has to use serializedActorPath